### ISSUE TITLE
yolov5 in the dlstreamer 2022.2

### ISSUE THREAD

**Issue by howarwang** [dlstreamer/dlstreamer#320]
I pulled the [latest image ](https://hub.docker.com/layers/intel/dlstreamer/2022.2.0-ubuntu20-gpu815-devel/images/sha256-9a749f33058abfc8453a0b11b83c511b7d258d3314522f37319bcc7ca2eccb07?context=explore) and created the container.

In the container, I tested yolov5s (version: v6.2, dataset: coco ) in gst_launch and but failed. The OpenVINO model(xml.bin) works well in OpenVINO env.

**Part of script:**

DETECTION_MODEL=/models/object-recognition/yolov5s/yolov5s_coco.xml
DETECTION_MODEL_PROC=/opt/intel/dlstreamer/samples/gstreamer/model_proc/public/yolo-v5.json
LABELS=/opt/intel/dlstreamer/samples/labels/coco_80cl.txt

PIPELINE="gst-launch-1.0 \
  ${SOURCE_ELEMENT} ! ${DECODE_ELEMENT} ! queue ! \
  gvadetect model=$DETECTION_MODEL \
            model-proc=$DETECTION_MODEL_PROC \
            labels-file=$LABELS\
            inference-interval=${DETECTION_INTERVAL} \
            threshold=0.4 \
            device=${DEVICE} ! \
  queue ! \
  gvatrack tracking-type=${TRACKING_TYPE} ! \
 
  queue ! \
  $SINK_ELEMENT"

**Error:**

ERROR: from element /GstPipeline:pipeline0/GstGvaDetect:gvadetect0: base_inference based element initialization has been failed.
Additional debug info:
/root/gst-video-analytics/gst/inference_elements/base/gva_base_inference.cpp(785): gva_base_inference_set_caps (): /GstPipeline:pipeline0/GstGvaDetect:gvadetect0:

post-processing is NULL.
Execution ended after 0:00:00.404554700
Setting pipeline to NULL ...
ERROR: from element /GstPipeline:pipeline0/GstGvaDetect:gvadetect0: base_inference failed on frame processing
Additional debug info:
/root/gst-video-analytics/gst/inference_elements/base/gva_base_inference.cpp(935): gva_base_inference_transform_ip (): /GstPipeline:pipeline0/GstGvaDetect:gvadetect0:

Failed to submit images to inference
        Mapper is null
ERROR: from element /GstPipeline:pipeline0/GstURISourceBin:urisourcebin0/GstSoupHTTPSrc:souphttpsrc0: Internal data stream error.
Additional debug info:
../libs/gst/base/gstbasesrc.c(3127): gst_base_src_loop (): /GstPipeline:pipeline0/GstURISourceBin:urisourcebin0/GstSoupHTTPSrc:souphttpsrc0:
streaming stopped, reason error (-5)
Freeing pipeline ...

BTW, yolov4 works well in the same script. 

Do you have official example to do yolov5 inference? Thanks.

**Comment by howarwang** [comment]
@mikhail-nikolskiy  Can you please check if I use it incorrectly

**Comment by howarwang** [comment]
BTW, I cloned yolov5 repo to dlstreamer container and tested detect.py using openvino. The detection results look good.

**Comment by brmarkus** [comment]
@howarwang which YoloV5 model have you tested? Have you tested `yolov5s`?
Can you provide details and share what you did and how it looked like, please?

**Comment by howarwang** [comment]
@brmarkus , I tested official yolov5s (coco80) which works well in detect.py from yolov5 repo.

**Comment by brmarkus** [comment]
@howarwang yes, the model is working with the YoloV5 Github repo's test-file `detect.py`.

Could you check whether the model can be used with DL-Streamer, too, please?

**Comment by howarwang** [comment]
@brmarkus  The model can work in openvino 2022.1 but doesnt work in dlstreamer 2022.2.
The same gst_launch and python pipeline can use yolov4 to inference, but if switch to yolov5, errors triggered like above.

**Comment by vidyasiv** [comment]
Hello @howarwang , 
I ran with docker image [intel/dlstreamer:2022.2.0-ubuntu20-gpu815-devel](https://hub.docker.com/layers/intel/dlstreamer/2022.2.0-ubuntu20-gpu815-devel/images/sha256-9a749f33058abfc8453a0b11b83c511b7d258d3314522f37319bcc7ca2eccb07?context=explore) and with this [model](https://github.com/dlstreamer/pipeline-zoo-models/tree/main/storage/yolov5s-416/FP32) with device=CPU on a TGL host and was able to see inference results.

Pipeline I ran with :

```
DETECTION_MODEL=$BASE/FP32/yolov5s.xml
DETECTION_MODEL_PROC=/opt/intel/dlstreamer/samples/gstreamer/model_proc/public/yolo-v5.json
LABELS=/opt/intel/dlstreamer/samples/labels/coco_80cl.txt
gst-launch-1.0 filesrc location=SAMPLE_MEDIA ! decodebin force-sw-decoders=true \
! gvadetect model=$DETECTION_MODEL model-proc=$DETECTION_MODEL_PROC labels-file=$LABELS threshold=0.4 \
! queue ! gvatrack tracking-type=zero-term-imageless \
! queue ! gvametaconvert ! gvametapublish file-format=json-lines ! fakesink 
```

Model was prepared per these steps : https://dlstreamer.github.io/dev_guide/yolov5_model_preparation.html 

Could you share more details about the model you used, pipeline parameters set, whether it was on CPU/GPU etc?

**Comment by howarwang** [comment]
@vidyasiv  Thanks Vidya.
I tired your model and it works!  I just remembered my model(xml,bin) is directly converted from yolov5.pt instead of to onnx and then to opennivo IR. Also my yolov5 version is V6.2. I am gonna try to convert my yolov5.pt(v6.2) to onnex and then to openvino IR. Will update here soon. Thanks again.

**Comment by vidyasiv** [comment]
@howarwang , please also refer to https://github.com/dlstreamer/dlstreamer/issues/322 i.e if you plan to use a different input size, the model-proc will need cells_number updated.

**Comment by howarwang** [comment]
Thanks @vidyasiv. I changed cell_number based on the input size.
